### PR TITLE
Translate the database.en.md from database.cn.md

### LINKED ISSUE #6070: [Translation] Translate technical articles concerning Apache ShardingSphere
Hi, community,

This issue is to call for volunteers to **translate some technical articles from Chinese into English**. Anyone who enjoys translation or wants to make some contributions to open source community is warmly welcomed! 👏 

### Competencies needed
1. English translation skill
2. Basically, know what ShardingSphere is doing

### Article list
- [x] 1. [Start ElasticJob off with new goal](https://mp.weixin.qq.com/s/XJFj_vOj3bR6HRQUzy1ikg) --> @rudy2steiner
- [x] 2. [How to construct the distributed database](https://mp.weixin.qq.com/s/YSoX0Apq6vQeu8TU_53EAw) --> @EasonFeng5870
- [x] 3. [Apache ShardingSphere is included in CNCF Landscape](https://mp.weixin.qq.com/s/hAxON1CMYvgFf6sFIHtDHA) --> @mlw12345678
- [x] 4. [The mixed open-source distributed transaction solution](https://mp.weixin.qq.com/s/fYESyuw7O4u7wB-tSWlhyg) --> @mlw12345678
- [x] 5. [Would you like to become an Apache committer](https://mp.weixin.qq.com/s/sFqOz10KTk23IhmmDTy67g) --> @mlw12345678
- [x] 6. [How automatic executor of ShardingSphere works](https://mp.weixin.qq.com/s/hIKZPJDfGfeduW4frwVN_w)
- [x] 7. [How to merge the child resultsets](https://mp.weixin.qq.com/s/5__SGXEdvGqkVhpnrW6zug)
- [ ] 8. [The quick explanation of ShardingSphere transaction module](https://mp.weixin.qq.com/s/0LtqRTL8HewUAdZCk8LRLg)
- [x] 9. [What is the ShardingProxy](https://mp.weixin.qq.com/s/bTX0n9tH4x8eue19b7rdGw) --> @wcsoft 

### How to submit your translation
1. Add your translation to `xxx.en.md` in [blog](https://github.com/apache/shardingsphere/tree/master/docs/blog/content/material).

2. Raise a PR for review.

### Notice
1. All the Chinese articles named `xxx.cn.md` could be found in the same directory of Your translation as well. 
2. If you want o find a partner to work together for these long articles, please be free to give `recruit comment` below. 😉 
3. You can give a brief translation for these author profiles in the articles. Like just translate their names and titles.

If you want to have a try, **please comment here with** `Article number and your estimated deadline`. 

Besides, **any feedback and comments are welcomed as well**! 😃

### PR DESCRIPTION
**Pull request by EasonFeng5870**
Ref #6070 No2

Changes proposed in this pull request:
- support english version.

### CODE REVIEW

File: docs/blog/content/material/database.en.md [hunk 1]
```diff
@@ -4,4 +4,296 @@ weight = 6
 chapter = true
 +++
 
-## TODO
+## How we build a distributed database
+
+Author | Liang Zhang
+
+  In the past few decades, relational database continuously occupy an absolute dominant position in the field of databases. The advantages of relation database, like stability, safety, ease of use that are the cornerstone of a modern system. With the rapid development of the Internet, databases built on stand-alone systems are no longer able to meet higher and higher concurrent requests and larger and larger data storage requirements. Therefore, distributed databases are more and more widely adopted.
```
Comment (wubingting) [inline docs/blog/content/material/database.en.md:11]: The advantages of relational database, like stability, safety, ease of use that are the cornerstone of a modern system.

**Comment by tristaZero** [comment]
@EasonFeng5870 Hi Thanks for your PR, @terrymanu will take charge of the review, please give him some time. :)

File: docs/blog/content/material/database.en.md [hunk 2]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+#### Distributed transaction
+
+Transaction is the core function of the database system. Distributed uncertainty and transaction complexity determine that there is no one-size-fits-all solution in the field of distributed transactions.
+
+Facing the current situation of blooming flowers, Apache ShardingSphere provides a highly open solution that uses standard interfaces to unify and integrate third-party distributed transaction frameworks independently selected by developers to meet the application requirements of various scenarios. In addition, Apache ShardingSphere also provides a new distributed transaction solution JDTX to make up for the lack of existing solutions.
+
+Standardized integrated interface
+
+Apache ShardingSphere provides a unified adaptation interface for local transactions, two-phase transactions, and flexible transactions, and docks with a large number of existing third-party mature solutions. Through the standard interface, developers can easily integrate other integration solutions into the Apache ShardingSphere platform.
```
Comment (terrymanu) [inline docs/blog/content/material/database.en.md:107]: `flexible transactions` should be `BASE transactions`

File: docs/blog/content/material/database.en.md [hunk 3]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+![](https://shardingsphere.apache.org/blog/img/database9.jpg)
+
+Its design feature is to separate the data in the transaction (called active data) from the data that is not in the transaction (called placement data). After the active data is placed on the WAL, it is saved in the form of KV to the MVCC memory engine. Placed data is synchronized to the final storage medium (such as a relational database) by asynchronously flashing the REDO logs in WAL in a flow controllable manner. The transactional memory query engine is responsible for retrieving relevant data from the active data in KV format using SQL, merging it with the data on the market, and obtaining the data version that is visible to the current transaction according to the transaction isolation level.
+
+JDTX reinterprets the database transaction model with a new architecture. The main highlights are:
+
+**1. The internalized distributed transaction is a local transaction**
```
Comment (terrymanu) [inline docs/blog/content/material/database.en.md:138]: It is better if change `The internalized distributed transaction is a local transaction` to `Convert  distributed transactions to local one`

File: docs/blog/content/material/database.en.md [hunk 4]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+JDTX's MVCC engine is a centralized cache. It can internalize the two-phase commit to the one-phase commit to maintain the atomicity and consistency of the data in a single node, that is, reduce the scope of distributed transactions to the scope of local transactions. JDTX guarantees the atomicity and consistency of transaction data by ensuring that all access to transaction data passes through the active data of the MVCC engine + the final data-end data combination.
+
+**2. Non-destructive transaction isolation mechanism**
```
Comment (terrymanu) [inline docs/blog/content/material/database.en.md:142]: It is better to change `Non-destructive transaction isolation mechanism` to `Fully support all transaction isolation levels`

File: docs/blog/content/material/database.en.md [hunk 5]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+The method of asynchronously flashing active data to the storage medium greatly improves the upper limit of data writing performance. Its performance bottleneck has shifted from the time of writing to the database to the time of placing it to WAL and MVCC engine.
+
+Similar to the WAL system of the database, the WAL of JDTX also adopts the way of sequential log appending, so it can be simply understood that the time-consuming WAL of JDTX = the time-consuming WAL of the database system. The MVCC engine uses a KV data structure, which takes less time to write than a database that maintains BTree indexes. Therefore, the upper limit of the data update performance of JDTX can even be higher than that of no transaction.
+
+**4. High availability**
+
+Both WAL and MVCC engines can maintain high availability and horizontal scalability through active and standby and sharding. When the MVCC engine is completely unavailable, the data in WAL can be synchronized to the database through the recovery mode to ensure the integrity of the data.
+
+**5. Cross-multiple database transactions**
```
Comment (terrymanu) [inline docs/blog/content/material/database.en.md:156]: change `Cross-multiple database transactions` to `Support transactions between different databases`

File: docs/blog/content/material/database.en.md [hunk 6]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+Desensitization
+
+Apache ShardingSphere allows users to automatically encrypt the specified data column and store it in the database without modifying the code, and decrypt it when the application obtains the data to ensure the security of the data. When the data in the database is leaked inadvertently, the sensitive data information is completely encrypted, so it will not cause greater security risks.
+
+Shadow Schema Table
+
+Apache ShardingSphere can automatically route user-marked data to the shadow schema (table) when the system performs a full link pressure test. The shadow schema (table) pressure measurement function can make online pressure measurement a normal state, and users do not need to care about the cleaning of pressure measurement data. This function is also under high-speed incubation.
+
+### 4 Route planning
```
Comment (terrymanu) [inline docs/blog/content/material/database.en.md:243]: `Route planning` should be `Roadmap`

File: docs/blog/content/material/database.en.md [hunk 7]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+As you can see, Apache ShardingSphere is on the track of rapid development, and more and more functions that have no strong relationship with the "sub-database and sub-table" were added to it. But the functions of these products are not obtrusive, but they can help Apache ShardingSphere become a more diversified distributed database solution. Apache ShardingSphere will focus on the following lines in the future.
+
+#### Pluggable platform
+
+More and more scattered functions need to be further sorted out. The existing architecture of Apache ShardingSphere is not enough to fully absorb such a wide range of product functions. The flexible functional pluggable platform is the adjustment direction of Apache ShardingSphere's future architecture.
+
+The pluggable platform completely disassembles Apache ShardingSphere from both technical and functional aspects. The panoramic view of Apache ShardingSphere is as follows:
```
Comment (terrymanu) [inline docs/blog/content/material/database.en.md:251]: `panoramic view` should be `landscape`

File: docs/blog/content/material/database.en.md [hunk 8]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+#### Query engine
```
Comment (terrymanu) [inline docs/blog/content/material/database.en.md:259]: `Query engine` should be `Query optimizer`

File: docs/blog/content/material/database.en.md [hunk 9]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+At present, Apache ShardingSphere only distributes SQL to the corresponding database through correct routing and rewriting to manipulate the data. The query engine that calculates and issues the database that can be fully utilized, but cannot effectively support complex related queries and subqueries. The SQL on KV query engine based on relational algebra has become mature with the development of JDTX, and its accumulated experience is fed back to the SQL query engine, which can enable Apache ShardingSphere to better support complex queries such as subqueries and cross-database related queries.
+
+#### Multiple data copies
```
Comment (terrymanu) [inline docs/blog/content/material/database.en.md:263]: `Multiple data copies` should be `Multiple replica`

File: docs/blog/content/material/database.en.md [hunk 10]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+With the expansion of the Apache ShardingSphere ecosystem, the status of the project controlled by a few developers has long been broken. The current Apache ShardingSphere has nearly one hundred contributors and nearly 20 core committers, who have jointly created this community that follows the Apache Way. Apache ShardingSphere is a standard Apache Software Foundation open source project and is not controlled by a commercial company or a few core developers.
+
+At present, more than 100 companies have clearly stated that they are using Apache ShardingSphere, and readers can find the user wall that adopts the company from the official website.
+
+As the community matures, Apache ShardingSphere grows faster and faster. We sincerely invite interested developers to participate in the Apache ShardingSphere community to improve the expanding ecosystem.
+
+**project address:**
+
+https://github.com/apache/incubator-shardingsphere
```
Comment (terrymanu) [inline docs/blog/content/material/database.en.md:291]: `https://github.com/apache/incubator-shardingsphere`  can rename to `https://github.com/apache/shardingsphere`

File: docs/blog/content/material/database.en.md [hunk 11]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+### 6 About the author
+
+Zhang Liang, head of data research and development at JD.com, initiator of Apache ShardingSphere & PPMC, head of JDTX.
```
Comment (terrymanu) [inline docs/blog/content/material/database.en.md:295]: `PPMC` should be `PMC`

File: docs/blog/content/material/database.en.md [hunk 12]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
 +++
 
-## TODO
+## How we build a distributed database
+
+Author | Liang Zhang
+
+  In the past few decades, relational database continuously occupy an absolute dominant position in the field of databases. The advantages of relational database, like stability, safety, ease of use that are the cornerstone of a modern system. With the rapid development of the Internet, databases built on stand-alone systems are no longer able to meet higher and higher concurrent requests and larger and larger data storage requirements. Therefore, distributed databases are more and more widely adopted.
+
+  All along, the database field has been dominated by Western technology companies and communities. Nowadays, more and more domestic database solutions take distributed as the fulcrum, and gradually make achievements in this field. Apache ShardingSphere is one of the distributed database solutions and the only database middleware in the Apache Software Foundation so far.
```
[the remark this answers is not in the document]
Comment (EasonFeng5870) [reply docs/blog/content/material/database.en.md:13]: noted.

File: docs/blog/content/material/database.en.md [hunk 13]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+- Distributed storage: Data storage is not limited by the disk capacity of a single machine, and the storage capacity can be improved by increasing the number of data servers;
+    
+- Separation of computing and storage: Computing nodes are stateless and can increase computing power through horizontal expansion. Storage nodes and computing nodes can be optimized hierarchically;
+  
+- Distributed transaction: A high-performance, distributed transaction processing engine that fully supports the original meaning of local transactions ACID;
+    
+- Elastic scaling: You can dynamically expand and shrink data storage nodes anytime, anywhere without affecting existing applications;
+    
+- Multiple data copies: Automatically copy the data to multiple copies across datacenters in a strong and consistent manner to ensure the absolute security of the data;
```
[the remark this answers is not in the document]
Comment (EasonFeng5870) [reply docs/blog/content/material/database.en.md:27]: great.

File: docs/blog/content/material/database.en.md [hunk 14]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+Positioned as a cloud-native database agent for Kubernetes, it will proxy all access to the database in the form of Sidecar. Through a non-central, zero-intrusion solution, it provides the meshing layer that interacts with the database, called as Database Mesh, which can also be called a database grid.
+
+![](https://shardingsphere.apache.org/blog/img/database3.jpg)
+
+**Hybrid architecture with separate computing and storage**
+
+Sharding-JDBC adopts a decentralized architecture, suitable for high-performance lightweight OLTP applications developed in Java; Sharding-Proxy provides static entry and heterogeneous language support, suitable for OLAP applications and management and operation and maintenance of sharded databases scene.
+
+Each architecture solution has its own advantages and disadvantages. The following table compares the advantages and disadvantages of various architecture models in different scenarios:
```
[the remark this answers is not in the document]
Comment (EasonFeng5870) [reply docs/blog/content/material/database.en.md:60]: noted.

File: docs/blog/content/material/database.en.md [hunk 15]
```diff
@@ -4,4 +4,296 @@ weight = 6
...
+
+![](https://shardingsphere.apache.org/blog/img/database6.jpg)
+
+The main process is as follows:
+
+1. Obtain the SQL and parameters input by the user by parsing the database protocol package or JDBC driver;
+    
+2. Parse SQL into AST (Abstract Syntax Tree) according to lexical analyzer and grammar analyzer, and extract the information required for fragmentation;
+    
+3. Match the shard key according to the user preset algorithm and calculate the routing path;
```
[the remark this answers is not in the document]
Comment (EasonFeng5870) [reply docs/blog/content/material/database.en.md:88]: releated also changed.

File: docs/blog/content/material/database.en.md (hunk 2 above)
Comment (EasonFeng5870) [reply docs/blog/content/material/database.en.md:107]: related also be replaced.

**Comment by EasonFeng5870** [comment]
@terrymanu， thanks for the revew.

**Pull request merged**